src/components/abstractComponent.ts

```diff
@@ -226,7 +223,7 @@ export module Component {
      * @param {number} [availableHeight] - the height of the container element
      * @returns {Component} The calling component.
      */
-    public resize(width?: number, height?: number): AbstractComponent {
+    public forceRedraw(width?: number, height?: number): AbstractComponent {
```

Comment:
We could take advantage of the optional variable signature here.  If any of width or height are set then the rootSVG size has to be set.  Otherwise just redraw.  I dunno if a separate API is necessary unless you think the new API will do something fundamentally different.